### PR TITLE
[bug_fix] Fix importing for mso_schema_site to support multiple templates in the same schema with the same site (DCNE-203)

### LINKED ISSUE #295: mso_schema_site import not supporting multiple templates (DCNE-203)
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.7.3

### MSO version

* V 4.2.3k

### APIC version and APIC Platform for Site Level Resources

* V 5.2/6.0

### Affected Resource(s)

* mso_schema_site

### Terraform Configuration Files

```hcl
resource "mso_schema" "schema1" {
    name = "schema1"
}

resource "mso_schema_site" "T1" {
}

terraform import mso_schema_site.T1 66f275b586669d8c4c40deb6/site/ACI_PROD_1
```

### Expected Behavior

In a situation where schema has multiple templates, and more than one template is assigned to the same site, it is not possible to import resource with the correct template. The import command expects schema ID and a site name but doesn't allow to provide template. We should be able to import that resource for each template 

### Actual Behavior

Only one template assigned to a site would be imported.

### Steps to Reproduce

* #0000

### PR DESCRIPTION
**Pull request by akinross**
fixes #295

### CODE REVIEW

File: mso/resource_mso_schema_site.go [hunk 1]
```diff
@@ -106,21 +108,31 @@ func resourceMSOSchemaSiteImport(d *schema.ResourceData, m interface{}) ([]*sche
...
-		if apiSiteId == stateSiteId {
+		if apiSiteId == stateSiteId && apiTemplate == template {
+			// d.SetId(fmt.Sprintf("%s/sites/%s-%s", schemaId, stateSiteId, template))
 			d.SetId(apiSiteId)
 			d.Set("schema_id", schemaId)
 			d.Set("site_id", apiSiteId)
 			d.Set("template_name", apiTemplate)
+
+			// when undeploy var is provided set it else set default value
+			if len(get_attribute) == 7 {
```
Comment (samiib) [inline mso/resource_mso_schema_site.go:119]: Should there also be a check that the 6th attribute is actually `undeploy_on_destroy`?

File: mso/resource_mso_schema_site.go [hunk 2]
```diff
@@ -106,21 +108,31 @@ func resourceMSOSchemaSiteImport(d *schema.ResourceData, m interface{}) ([]*sche
 		apiSiteId := models.StripQuotes(tempCont.S("siteId").String())
 		apiTemplate := models.StripQuotes(tempCont.S("templateName").String())
 
-		if apiSiteId == stateSiteId {
+		if apiSiteId == stateSiteId && apiTemplate == template {
+			// d.SetId(fmt.Sprintf("%s/sites/%s-%s", schemaId, stateSiteId, template))
```
Comment (samiib) [inline mso/resource_mso_schema_site.go:112]: Should the commented out code be removed?

File: mso/resource_mso_schema_site.go [hunk 3]
```diff
@@ -173,6 +185,7 @@ func resourceMSOSchemaSiteRead(d *schema.ResourceData, m interface{}) error {
 		apiTemplate := models.StripQuotes(tempCont.S("templateName").String())
 
 		if apiSiteId == stateSiteId && apiTemplate == stateTemplate {
+			// d.SetId(fmt.Sprintf("%s/sites/%s-%s", schemaId, stateSiteId, stateTemplate))
```
Comment (samiib) [inline mso/resource_mso_schema_site.go:188]: Should the commented out code be removed?

**Review by samiib — CHANGES_REQUESTED** [review]

File: mso/resource_mso_schema_site.go (hunk 1 above)
Comment (akinross) [reply mso/resource_mso_schema_site.go:119]: Does not matter, could be any arbitrary string as long as the string after is a boolean parsable string. This is similar to current behaviour of importing commands

File: mso/resource_mso_schema_site.go (hunk 2 above)
Comment (akinross) [reply mso/resource_mso_schema_site.go:112]: actually left the comment there on purpose that this is the id we would actually want, but I did not want to introduce a breaking change into the resource.

File: mso/resource_mso_schema_site.go (hunk 3 above)
Comment (akinross) [reply mso/resource_mso_schema_site.go:188]: actually left the comment there on purpose that this is the id we would actually want, but I did not want to introduce a breaking change into the resource.

**Review by gmicol — APPROVED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Review by samiib — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**